From 7b078765a6a6f28e96d172e5e8884dbee7fa1e56 Mon Sep 17 00:00:00 2001
From: Johannes Thoma <johannes@johannesthoma.com>
Date: Tue, 26 Sep 2017 20:57:30 +0200
Subject: [PATCH 01/15] Call windrbd_set_device_active on successful attach.

We also need it on drbd_adm_connect in case there is no attach.
---
 drbd/drbd_nl.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drbd/drbd_nl.c b/drbd/drbd_nl.c
index ee6babf..7e0f75d 100644
--- a/drbd/drbd_nl.c
+++ b/drbd/drbd_nl.c
@@ -45,6 +45,7 @@
 #include <linux/kthread.h>
 #include <linux/security.h>
 #include <net/genetlink.h>
+#include "drbd_windows.h"
 
 /* .doit */
 // int drbd_adm_create_resource(struct sk_buff *skb, struct genl_info *info);
@@ -2828,6 +2829,8 @@ int drbd_adm_attach(struct sk_buff *skb, struct genl_info *info)
 	put_ldev(device);
 	mutex_unlock(&resource->adm_mutex);
 	drbd_adm_finish(&adm_ctx, info, retcode);
+
+	windrbd_set_drbd_device_active(device, 1);
 	return 0;
 
  force_diskless_dec:
-- 
2.8.0-rc4

